src/server/raft_client.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::net::SocketAddr;
+use futures::{future, Future, Stream};
+use futures::sync::mpsc;
+use tokio_core::reactor::Handle;
+use grpc::error::GrpcError;
+use kvproto::raft_serverpb::RaftMessage;
+use kvproto::tikvpb_grpc::{TiKVAsync, TiKVAsyncClient};
+use util::worker::FutureRunnable;
+use util::{HashMap, TryInsertWith};
+use super::Result;
+
+// SendTask delivers a raft message to other stores.
+pub struct SendTask {
+    pub addr: SocketAddr,
+    pub msg: RaftMessage,
+}
+
+impl fmt::Display for SendTask {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "send raft message to {:?}", self.addr)
+    }
+}
+
+struct Conn {
+    _client: TiKVAsyncClient,
```

Comment:
We create a stream for each Client to send raft messages, so we don't have to wait response, compare to unary rpc call.